common/flatpak-run.c

```diff
@@ -863,8 +863,23 @@ flatpak_run_in_transient_unit (const char *appid, GError **error)
   if (!manager)
     goto out;
 
-  appid_escaped = systemd_unit_name_escape (appid);
-  name = g_strdup_printf ("app-flatpak-%s-%d.scope", appid_escaped, getpid ());
+  app_id_escaped = systemd_unit_name_escape (app_id);
+
+  if (instance_id)
+    {
+      g_autofree char *instance_id_escaped = NULL;
+
+      instance_id_escaped = systemd_unit_name_escape (instance_id);
+      name = g_strdup_printf ("app-flatpak-%s-instance%s.scope",
+                              app_id_escaped,
+                              instance_id_escaped);
+    }
+  else
+    {
+      name = g_strdup_printf ("app-flatpak-%s-internal%d.scope",
```

Comment:
If we do keep calling `flatpak_run_in_transient_unit()` for extra-data, I'd be tempted to do this as `app-flatpak-%s-i%s.scope` and `app-flatpak-%s-p%d.scope`, or replace `i` and `p` with `inst` and `pid`, or something - these strings are already somewhat long.